### DISCUSSION TITLE
Missing webresource files while running Cypress which then throws an error (uncaught exception)ReferenceError: Page_ClientValidate is not defined  after a click action

### DISCUSSION THREAD

**Discussion by Walczynska** [cypress-io/cypress#26874] [Questions and Help]
Many Webresource files are missing when I try to run my Cypress test. The error (uncaught exception)ReferenceError: Page_ClientValidate is not defined appears when my test tries to click a button. I have tried the following solution
   Cypress.on('uncaught:exception', (err, runnable) => {
    return false});
 but the button still doesn't click. After a review I noticed that the Websource files where the Page_ClientValidate is defined are missing when I run the test in Cypress comparing to when I perform the action manually on the page. (See attached screenshots)
Files on the page when manually clicking on the Save Address button
![image](https://github.com/cypress-io/cypress/assets/56719178/a6ee190d-fa42-49cb-bb54-2f308daa383b)
The files in Cypress:
![image](https://github.com/cypress-io/cypress/assets/56719178/31c659c7-0c17-40ab-9ac1-21f32c7323ca)

![image](https://github.com/cypress-io/cypress/assets/56719178/20cd47a6-72fc-4e03-8725-3b6e2af8f0b0)

![image](https://github.com/cypress-io/cypress/assets/56719178/00f3bce6-5c28-4009-8f6b-4ead305708ff)

Even if I pause the test and manually click the button the same error appears.

**Comment by portyu9** [comment]
Hi @Walczynska 

`Page_ClientValidate` is an ASP.NET Web Forms function supplied by `WebResource.axd`/`ScriptResource.axd`. Since those resources are missing only when the page is opened through Cypress, the issue is most likely ASP.NET browser/User-Agent detection, not `cy.click()`.

Set Cypress to use the same User-Agent as the browser where the page works:

```js
// cypress.config.js
module.exports = {
  e2e: {
    userAgent: 'YOUR_NORMAL_CHROME_USER_AGENT'
  }
}
```

Or first try running the test in Chrome:

```bash
npx cypress open --browser chrome
```

If you control the ASP.NET application, the better server-side fix is to update its browser definitions or configure the page as an `uplevel` client so Web Forms always emits the required client-validation scripts.

Do not use `Cypress.on('uncaught:exception', () => false)` as the fix—it only suppresses the error while `Page_ClientValidate` remains missing.
